src/cli/npc.rs

```diff
@@ -1,35 +1,59 @@
-use nispor::get_state;
+use clap::{clap_app, crate_authors, crate_version};
+use nispor::NetState;
 use serde_json;
-use std::env::args;
+use std::process;
 
 fn main() {
-    let argv: Vec<String> = args().collect();
-    match get_state() {
+    let matches = clap_app!(npc =>
+        (version: crate_version!())
+        (author: crate_authors!())
+        (about: "Nispor CLI")
+        (@subcommand link =>
+            (about: "Network status of an interface")
+            (@group interface =>
+                (@arg ifname: [INTERFACE_NAME] "interface name")
+            )
+        )
+        (@subcommand route =>
+            (about: "Show routes")
+        )
+    )
+    .get_matches();
+
+    let result = match NetState::new() {
         Ok(state) => {
-            if argv.len() > 1 {
-                match &state.ifaces.get(&argv[1]) {
-                    Some(iface) => println!(
-                        "{}",
-                        serde_json::to_string_pretty(iface).unwrap()
-                    ),
-                    None => {
-                        if argv[1] == "route" {
-                            println!(
-                                "{}",
-                                serde_json::to_string_pretty(&state.routes)
-                                    .unwrap()
-                            );
-                        } else {
-                            eprintln!("Interface '{}' not found", argv[1]);
-                        }
+            if let Some(matches) = matches.subcommand_matches("link") {
+                if let Some(ifname) = matches.value_of("ifname") {
+                    if let Some(iface) = &state.ifaces.get(ifname) {
+                        serde_json::to_string_pretty(iface)
+                            .map_err(|e| format!("{}", e))
+                    } else {
+                        Err(format!("interface '{}' not found", ifname))
                     }
+                } else {
+                    serde_json::to_string_pretty(&state.ifaces)
+                        .map_err(|e| format!("{}", e))
                 }
+            } else if let Some(_) = matches.subcommand_matches("route") {
+                serde_json::to_string_pretty(&state.routes)
+                    .map_err(|e| format!("{}", e))
             } else {
-                println!("{}", serde_json::to_string_pretty(&state).unwrap());
+                /* Show everything if no cmdline arg has been supplied */
+                serde_json::to_string_pretty(&state)
+                    .map_err(|e| format!("{}", e))
             }
         }
+        Err(e) => Err(format!("{}", e)),
+    };
+
+    match result {
+        Ok(json) => {
+            println!("{}", json);
+            process::exit(0);
+        }
         Err(e) => {
             eprintln!("{}", e);
+            process::exit(1);
```

Comment:
How about we print error in json format also?

Reply:
How about now?